### PR TITLE
Feat/reausable2

### PR DESCRIPTION
**Pull request by paologallinaharbur**

### CODE REVIEW

File: .github/workflows/on_prerelease.yaml [hunk 1]
```diff
@@ -147,80 +52,16 @@ jobs:
...
-      - name: Test uploaded package installability
-        uses: newrelic/integrations-pkg-test-action/linux@v1
-        with:
-          tag: ${{ env.TAG }}
-          integration: nri-${{ env.INTEGRATION }}
-          packageLocation: repo
-          stagingRepo: true
-          upgrade: false
+  pre-release:
+    needs: [package-win, test-integration-nix]
```
Comment (gsanchezgavier) [inline .github/workflows/on_prerelease.yaml:58]:
```suggestion
    needs: [test-integration-nix]
```

File: .github/workflows/on_push_pr.yaml [hunk 2]
```diff
@@ -87,21 +39,5 @@ jobs:
         uses: actions/setup-go@v4
         with:
           go-version-file: 'src/github.com/${{env.ORIGINAL_REPO_NAME}}/go.mod'
-      # One of the integration consumer tests is not working as expected to the environment bootstrap, to keep it
```
Comment (gsanchezgavier) [inline .github/workflows/on_push_pr.yaml:90]: why removeing the comment?

File: .github/workflows/on_push_pr.yaml [hunk 3]
```diff
@@ -7,70 +7,22 @@ on:
       - master
       - renovate/**
   pull_request:
+  workflow_dispatch:
 
 env:
-  TAG: "v0.0.0" # needed for goreleaser windows builds
-  REPO_FULL_NAME: ${{ github.event.repository.full_name }}
-  ORIGINAL_REPO_NAME: "newrelic/nri-kafka"
+  ORIGINAL_REPO_NAME: ${{ github.event.repository.full_name }}
```
Comment (gsanchezgavier) [inline .github/workflows/on_push_pr.yaml:13]: I think this is needed for the test job, if so I would move the env to that job so is clear that the other one doesn't need this

File: .github/workflows/on_prerelease.yaml [hunk 4]
```diff
@@ -8,128 +7,34 @@ on:
       - 'v*'
 
 env:
```
Comment (gsanchezgavier) [inline .github/workflows/on_prerelease.yaml:9]: I would move these env out of the global and to the specific jobs when they need it which I think is the extra test

**Review by gsanchezgavier — COMMENTED** [review]
I agree with the approach of leaving the special tests , just left some minor comments

File: .github/workflows/on_prerelease.yaml (hunk 1 above)
Comment (paologallinaharbur) [reply .github/workflows/on_prerelease.yaml:58]: Thanks for noticing! I am not sure how/why I was introducing this issue!

File: .github/workflows/on_prerelease.yaml (hunk 4 above)
Comment (paologallinaharbur) [reply .github/workflows/on_prerelease.yaml:9]: sure, done!

File: .github/workflows/on_push_pr.yaml (hunk 3 above)
Comment (paologallinaharbur) [reply .github/workflows/on_push_pr.yaml:13]: done!

**Review by gsanchezgavier — APPROVED** [review]
thanks!

**Pull request merged**